backend/src/sudokuboard.py

```diff
@@ -0,0 +1,102 @@
+from collections import deque
+from dataclasses import dataclass
+from functools import singledispatchmethod
+from itertools import islice
+from random import shuffle
+from typing import Final, List, Set
+
+
+@dataclass(eq=True, frozen=True)
+class Coord:
+    i: int = 0
+    j: int = 0
+
+
+@dataclass
+class SudokuCell:
+    pos: Final[Coord]
+    value: int = 0
+    SIZE: Final[int] = 3
+
+    def __post_init__(self):
+        self.neighbors: Set[Coord] = set()
+        for n in range(self.SIZE ** 2):
+            self.neighbors.add(Coord(n, self.pos.j))
+            self.neighbors.add(Coord(self.pos.i, n))
+
+        block_coord = self._get_block_coord()
+
+        for n in range(block_coord.i, block_coord.i + self.SIZE):
+            for m in range(block_coord.j, block_coord.j + self.SIZE):
+                self.neighbors.add(Coord(n, m))
+
+        self.neighbors.remove(self.pos)
+
+    def _get_block_coord(self) -> Coord:
+        return Coord(
+            (self.pos.i // self.SIZE) * self.SIZE,
+            (self.pos.j // self.SIZE) * self.SIZE,
+        )
+
+
+class SudokuBoard:
+    def __init__(self, SIZE=3):
+        self.SIZE = SIZE
+        self.VALID_VALUES = {i + 1 for i in range(SIZE ** 2)}
+        self.cells = deque()
```

Comment:
any particular reason why this is a `deque`?

Reply:
This is a leftover from the previous approach, I guess simple python `List` will do the work as well.

Reply:
`List` is actually faster. Moreover, cells are initialized by `list comprehension` from now on.